force-app/main/default/classes/AccountTriggerTest.cls

```diff
@@ -0,0 +1,326 @@
+@isTest
+public class AccountTriggerTest {
+  //more than 40,000 accounts have to be updated regularly
+  private static final Integer NUM_OF_TOP_LEVEL_ACCOUNTS = 4;
+  private static final Integer NUM_OF_CHILDREN_ACCOUNTS = 10;
+  private static final Decimal PARENT_DISCOUNT = 50;
+
+  @TestSetup
+  static void makeData() {
+    List<Account> topLevelAccounts = new List<Account>();
+
+    for (Integer i = 0; i < NUM_OF_TOP_LEVEL_ACCOUNTS; i++) {
+      Account topLevelAccount = new Account(
+        Name = 'Test Top Level Account' + i,
+        Discount__c = 20,
+        HasLockedDiscount__c = false
+      );
+      topLevelAccounts.add(topLevelAccount);
+    }
+    Test.startTest();
+    insert topLevelAccounts;
+    Test.stopTest();
+  }
+
+  @isTest
+  private static void itShouldUpdateDiscountOnUnlockedChildrenIfParentDiscountChanges() {
+    List<Account> topLevelAccounts = [
+      SELECT Id, Discount__c
+      FROM Account
+      LIMIT :NUM_OF_TOP_LEVEL_ACCOUNTS
+    ];
+
+    List<Account> childrenAccounts = itShouldCreateChildrenAccounts(
```

Comment:
test methods can start with 'itShould' but helper methods shouldn't. I understood you have helper and a test in one and the same method. 
Two methods would help
a test method itShouldCreateChildrenAccounts that executes and asserts a private `createChildrenAccounts` method

Reply:
Ok, so using Assert methods inside the helper method is it wrong? Should I delete them and only use them inside test methods?

Reply:
Its about separation of concerns. a method has one purpose ideally.  so i suggest to create a test method that tests the helper method and use the helper method in other test methods. Cf.

```
@isTest
private static void itShouldTestSomeAccountDiscountFunction() {

List<account> childAccounts = createChildAccounts(); 
/*logic and asserst*/
}


@isTest 
private static void itsShouldCreateChildAccounts() {

List<Account> accounts = createChildAccounts();
Assert.areEqual(200, accounts.size(), '200 children created');

}



private static List<Account> createChildAccounts() {
   /*logic*/
   return insertedChildAccounts;
}


```

Asserts in production code can be helpful, but personally I use them only during code writing and remove them again after a while. they are a better way to debug code and understand return values than system.debug